### ISSUE TITLE
is there some way to use pyaapt on batch of audio?

### ISSUE THREAD

**Issue by avishaiElmakies** [bjbschmitt/AMFM_decompy#14]
Hi, I would like to know if there is some way to add batch support for pyaapt (or tell me if i am missing something). I have a bunch of audio i need to get the f0 of. But i can't seem to get it to work with a batch of audio.

Thanks in advance!

**Comment by bjbschmitt** [comment]
Hi @avishaiElmakies.

By "extracting the f0 of a batch of audio" I believe that you mean that you want to perform the pitch estimation of multiple audio signals simultaneously, in order to speed the analysis of a huge speech dataset.

If this is the case, unfortunately this feature is in fact still not available.

Nevertheless, I have been working during the last months in a new set of features, which the final goal (I hope so) would be a improvement in the computation performance of the pyaapt. My idea is to test different resources available, like CUDA, parallelization, numba, etc, to check if any of them can increase the speed of the extraction.

Unfortunately (again), in the next AMFM_decompy release that I am currently finishing, these features will still not be available. My first milestone was establish a set of tools in order to "benchmark" the pyaapt algorithm (i.e. measure its f0 estimation accuracy and its computation time). 

This was a necessary step because, once I start altering the code, I need some kind of reference to check if my modifications are in fact improving the performance of the algorithm and not harming the f0 extraction accuracy. 

So, in summary, it may take some months until I reach my final goal (since I can only work on this project during my free time).

**Comment by avishaiElmakies** [comment]
@bjbschmitt I understand. Thanks for the detailed response!
